pkg/pgmodel/common/extension/extension.go

```diff
@@ -169,8 +178,9 @@ func MigrateExtension(conn *pgx.Conn, extName string, extSchemaName string, vali
 		_, err := connAlter.Exec(context.Background(),
 			fmt.Sprintf("ALTER EXTENSION %s UPDATE TO '%s'", extName,
 				getSqlVersion(newVersion, extName)))
+		// if migration fails, Do not crash just log an error. As there is an extension already present.
 		if err != nil {
-			return err
+			log.Error("msg", fmt.Sprintf("Failed to migrate extension %v from %v to %v: %v", extName, currentVersion, newVersion, err))
```

Comment:
ugh still not quite right I'm sorry. we can ignore upgrade failures only if the current version is still valid
```suggestion
                         if !validRange(currentVersion) {
			    return fmt.Errorf("The %v extension is not installed at the right version and the upgrades failed, need version: %v and the installed version is %s and the upgrade failed with %w", extName, rangeString, currentVersion, err)
			}
			log.Error("msg", fmt.Sprintf("Failed to migrate extension %v from %v to %v: %v", extName, currentVersion, newVersion, err))
```